clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -1088,6 +1088,9 @@ void CXXNameMangler::mangleNameWithAbiTags(GlobalDecl GD,
     return;
   }
 
+  while (DC->isRequiresExprBody())
+    DC = DC->getParent();
```

Comment:
Is this sufficient? For example, what if the lambda is still nested in a requires body, but its parent is some other kind of DeclContext?